packages/arb-token-bridge-ui/src/hooks/useArbQueryParams.tsx

```diff
@@ -97,6 +108,50 @@ export const AmountQueryParam = {
   }
 }
 
+// Parse chainId to ChainQueryParam or ChainId for orbit chain
+function encodeChainQueryParam(chainId: number | null | undefined) {
+  if (!chainId) {
+    return undefined
+  }
+
+  try {
+    const chain = getChainQueryParamForChain(chainId)
+    return chain.toString()
+  } catch (e) {
+    return undefined
+  }
+}
+
+// Parse ChainQueryParam/ChainId to ChainId
+// URL accept both chainId and chainQueryParam (string)
+function decodeChainQueryParam(
+  value: string | (string | null)[] | null | undefined
+): ChainId | undefined {
```

Comment:
Can we actually narrow to `ChainId` here or does it have to be `number`? I'm asking because of the custom chains as we can't dynamically update the enum

Reply:
The library only accept primitive types as return and input

Reply:
I was wrong. You can narrow down the output, but not the input.